pkg/traceql/engine_metrics.go

```diff
@@ -239,7 +241,7 @@ func (set SeriesSet) ToProtoDiff(req *tempopb.QueryRangeRequest, rangeForLabels
 
 			// todo: this loop should be able to be restructured to directly pass over
 			// the desired intervals
-			if ts < start || ts > end {
+			if ts < start || ts > end || math.IsNaN(value) {
```

Comment:
We need one more change which is to not output series which end up with no samples after filtering.  After this loop:

```go
		// Skip series with no samples after filtering
		if len(samples) == 0 {
			continue
		}
```

This handles cases like the following:

`{ } | min_over_time(span.foo)`  Currently returns a single series with no data points.

`{ } | min_over_time(span.http.status_code) by (span.http.status_code)` This is a contrived example, but it includes an empty `"nil"` series for the spans without http.status_code.

Reply:
Cool!